x-pack/plugins/observability/public/hooks/use_create_rule.ts

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useMutation } from '@tanstack/react-query';
+import { i18n } from '@kbn/i18n';
+import { BASE_ALERTING_API_PATH, RuleTypeParams } from '@kbn/alerting-plugin/common';
+import { v4 } from 'uuid';
+import { CreateRuleRequestBody } from '@kbn/alerting-plugin/common/routes/rule/apis/create';
+import { useKibana } from '../utils/kibana_react';
+
+export function useCreateRule<Params extends RuleTypeParams = never>() {
+  const {
+    http,
+    notifications: { toasts },
+  } = useKibana().services;
+
+  const createRule = useMutation<string, string, { rule: CreateRuleRequestBody<Params> }>(
+    ['createRule'],
+    ({ rule }) => {
+      try {
+        const ruleId = v4();
+        const body = JSON.stringify(rule);
+        return http.post(`${http.basePath.prepend(BASE_ALERTING_API_PATH)}/rule/${ruleId}`, {
+          body,
+        });
```

Comment:
i don't think we should be calling another plugin public API.
We should create a route and use AlertingClient exposed on server side to create an alert.

Reply:
@XavierM thoughts?

Reply:
If it's public, it should be ok to use them. We use at least 2 other RAC apis for fetching the rules and alerts.